### PR TITLE
Extend LVM class and test snapshot bigger than disk

### PR DESCRIPTION
**Pull request by shyRozen**
Signed-off-by: Shay Rozen <shay.rozen@gmail.com>

### CODE REVIEW

File: ocs_ci/ocs/cluster.py [hunk 1]
```diff
@@ -2230,6 +2249,150 @@ def get_lvm_thin_pool(self):
...
+                "data_percent": lv["data_percent"],
+                "metadata_percent": lv["metadata_percent"],
+                "move_pv": lv["move_pv"],
+                "mirror_log": lv["mirror_log"],
+                "copy_percent": lv["copy_percent"],
+                "convert_lv": lv["convert_lv"],
+            }
+            self.lv_data["lv_list"].append(lv["lv_name"])
+
+    def get_lv_data_percent_of_pvc(self, pvc_obj):
```
Comment (ebenahar) [inline ocs_ci/ocs/cluster.py:2299]: add docstring to all missing places

**Review by Avilir — APPROVED** [review]
left some comment which can be address in followup PR

File: ocs_ci/ocs/cluster.py [hunk 2]
```diff
@@ -2230,6 +2324,328 @@ def get_lvm_thin_pool(self):
...
+    def get_thin_pool1_size(self):
+        """
+        gets thin-pool size
+        Returns:
+            (str) thin pool size
+
+        """
+        return self.lv_data["thin-pool-1"]["lv_size"]
+
+    def get_thin_pool1_data_percent(self):
```
Comment (AviadP) [inline ocs_ci/ocs/cluster.py:2540]: missing docstring

File: ocs_ci/ocs/cluster.py [hunk 3]
```diff
@@ -2162,22 +2171,107 @@ class for lvm cluster
...
+                self.get_and_parse_lvs(),
+                self.get_and_parse_vgs(),
+                self.compare_thin_pool_data_percent(
+                    data_percent=0,
+                    sampler=True,
+                    timeout=60,
+                    fail=True,
+                    diff_allowed=0,
+                ),
+            ]
```
[the remark this answers is not in the document]
Comment (shyRozen) [reply ocs_ci/ocs/cluster.py]: Solved only I've used one extend in the end.

File: ocs_ci/ocs/cluster.py [hunk 4]
```diff
@@ -2162,22 +2171,107 @@ class for lvm cluster
...
+        Return:
+            (LVM) object
+        """
+        self.lv_data = None
+        self.lvmcluster = None
+        self.pv_data = None
+        self.version = None
+        self.ip = None
+        self.vg_data = None
+        func_list = []
```
[the remark this answers is not in the document]
Comment (shyRozen) [reply ocs_ci/ocs/cluster.py]: Thanks Avi. Moved to your suggestion

File: ocs_ci/utility/connection.py [hunk 5]
```diff
@@ -79,6 +81,9 @@ def exec_cmd(self, cmd):
         stdout = out.read().decode("ascii").strip("\n")
         stderr = err.read().decode("ascii").strip("\n")
         logger.debug(f"retcode: {retcode}")
-        logger.debug(f"stdout: {stdout}")
+        logger.info(f"stdout: {stdout}") if self.stdout else logger.debug(
+            f"stdout: {stdout}"
+        )
+        logger.info(f"stdout: {stdout}")
```
[the remark this answers is not in the document]
Comment (shyRozen) [reply ocs_ci/utility/connection.py]: Removed 87

File: ocs_ci/ocs/cluster.py (hunk 1 above)
Comment (shyRozen) [reply ocs_ci/ocs/cluster.py:2477]: Added.

**Review by ebenahar — APPROVED** [review]

**Review by Avilir — APPROVED** [review]

**Pull request merged**